### PR TITLE
Autoupdate ptroverride and use ptroverride in the reversefile export

### LINKED ISSUE #159: Oppdater PtrOverride når en ipaddresse ikke lengre er unik
Hver gang en ipadresse legges til eller slettes, så må man sjekke om ipadressen fortsatt/ikke lengre er unik og følgelig oppdatere PtrOverride.

Hvis en ipadresse er i bruk av host A, og en ny host B også vil benytte samme ipadresse, så vil man lage en PtrOverride som peker på host A.
Hvis host A slettes så fjernes overkjøringen, og følgelig vil reversen til ipadressen peke på host B.
Hvis en host C også peker på samme ipadresse, og er blitt opprettet før host A slettes, så opprettes ingen PtrOverride hvis host A slettes. Reversesonefileeksporten får heller klage for to som peker på samme, og at ingen PtrOverride eksisterer.

### PR DESCRIPTION
**Pull request by oyvindkolbu**
Resolves #159.

### CODE REVIEW

File: mreg/api/v1/tests.py [hunk 1]
```diff
@@ -249,10 +249,10 @@ def test_model_can_create_ptr(self):
     def test_model_can_change_ptr(self):
         """Test that the model is able to change a PTR Override."""
         clean_and_save(self.ptr_sample)
-        new_ptr = '129.240.202.124'
+        new_ptr = '10.0.0.3'
         self.ptr_sample.ipaddress = new_ptr
         clean_and_save(self.ptr_sample)
-        updated_ptr = PtrOverride.objects.filter(hostid__name='some-host.example.org')[0].ipaddress
+        updated_ptr = PtrOverride.objects.filter(hostid__name='host1.example.org')[0].ipaddress
```
Comment (ponas) [inline mreg/api/v1/tests.py]: `PtrOverride.objects.filter(hostid__name='host1.example.org').first().ipaddress`

File: mreg/api/v1/zonefile.py [hunk 2]
```diff
@@ -64,20 +67,41 @@ def generate(self):
             data += srv.zf_string(zone.name)
         return data
 
+def get_ipaddresses(network):
+    from_ip = str(network.network_address)
+    to_ip = str(network.broadcast_address)
+    where_str = "ipaddress BETWEEN '{}' AND '{}'".format(from_ip, to_ip)
```
Comment (ponas) [inline mreg/api/v1/zonefile.py]: https://docs.djangoproject.com/en/dev/ref/models/querysets/#range

File: mreg/api/v1/zonefile.py [hunk 3]
```diff
@@ -64,20 +67,41 @@ def generate(self):
             data += srv.zf_string(zone.name)
         return data
 
+def get_ipaddresses(network):
```
Comment (ponas) [inline mreg/api/v1/zonefile.py]: Could be property/method in the class `Zone`?

File: mreg/api/v1/zonefile.py [hunk 4]
```diff
@@ -86,39 +110,30 @@ def generate(self):
...
-        from_ip = str(network.network_address)
-        to_ip = str(network.broadcast_address)
-        where_str = "ipaddress BETWEEN '{}' AND '{}'".format(from_ip, to_ip)
-        ips = Ipaddress.objects.extra(where=[where_str],
-                                       order_by=["ipaddress"])
-        # XXX: need to check ptroverrides
-        return ips
-
     def generate(self, zone):
         network = get_network_from_zonename(zone.name)
```
Comment (ponas) [inline mreg/api/v1/zonefile.py]: `get_network_from_zonename` could be a property/method in the class `Zone`

File: mreg/api/v1/zonefile.py (hunk 2 above)
Comment (oyvindkolbu) [reply mreg/api/v1/zonefile.py]: Kan teste den... minnes det var noe brukne saker, men det var kanskje "__lt" og "__gt".

File: mreg/api/v1/zonefile.py [hunk 5]
```diff
@@ -133,7 +101,7 @@ def generate(self, zone):
         # TODO: delegated entries, if any
         data += ';\n; Delegations\n;\n'
         _prev_net = 'z'
-        for ip in get_ipaddresses(network):
+        for ip in zone.get_addresses():
```
Comment (ponas) [inline mreg/api/v1/zonefile.py]: `get_ipaddresses`

Comment (oyvindkolbu) [reply mreg/api/v1/zonefile.py]: Ops. fixed.

File: mreg/api/v1/zonefile.py (hunk 2 above)
Comment (oyvindkolbu) [reply mreg/api/v1/zonefile.py]: Range seems to be ok.

**Pull request merged**